store/searchlayer/layer.go

```diff
@@ -17,12 +17,14 @@ type SearchStore struct {
 	team         *SearchTeamStore
 	channel      *SearchChannelStore
 	post         *SearchPostStore
+	config       *model.Config
 }
 
-func NewSearchLayer(baseStore store.Store, searchEngine *searchengine.Broker) SearchStore {
+func NewSearchLayer(baseStore store.Store, searchEngine *searchengine.Broker, cfg *model.Config) SearchStore {
 	searchStore := SearchStore{
 		Store:        baseStore,
 		searchEngine: searchEngine,
+		config:       cfg,
```

Comment:
This doesn't handle config changes, I think, maybe another approach is to receive a config service here, and request the config each time. Other option is to add a "SetConfig" method and connect a ConfigListener to update this config every single time.

Reply:
Thanks @jespino! Added a config listener to update the configuration for the `SearchStore` in https://github.com/mattermost/mattermost-server/pull/14245/commits/9e3184fcee82772e556a175bf08f87b5e217f2a8. I'm happy to explore other approaches like the config service one if you think they might fit better